tabs/src/components/sample/QueryRunner.jsx

```diff
@@ -66,25 +68,62 @@ export function QueryRunner() {
         items: ['Key', 'Value'],
     };
 
-    const callGraph = () => {
-        // stub
-        console.log(requestType);
-        console.log(graphVersion);
-        console.log(query);
-        setResponseBody("");
-        setResponseHeaders([{ key: 'Content', items: ['Content', 'json'] }]);
-    };
+    async function callGraph() {
+        const queryParameters = query.substring(GRAPH_URL.length + graphVersion.length, query.length);
+        const url = RSC_API_URL + graphVersion + queryParameters;
+        const cleanedHeaders = {};
+        for (const i of requestHeaders) {
+            cleanedHeaders[i.items[0]] = i.items[1];
+        }
+        let requestParams = {};
+        if (requestType === requestTypes.GET) {
+            requestParams = {
+                method: requestType,
+                headers: cleanedHeaders,
+            };
+        } else {
+            requestParams = {
+                method: requestType,
+                headers: cleanedHeaders,
+                body: requestBody
+            };
+        }
+        const graphResponse = await fetch(url, requestParams);
+        let graphResponseHeaders = [];
+        for (const p of graphResponse.headers.entries()) {
+            graphResponseHeaders.push({
+                key: p[0],
+                items: [p[0], p[1]]
+            });
+        }
+        setResponseHeaders(graphResponseHeaders);
+        setReponseState(graphResponse.status + " " + graphResponse.statusText);
+        const text = await graphResponse.json();
+        setResponseBody(JSON.stringify(text, undefined, 4));
+    }
```

Comment:
In our cleanup stories, we should have tasks to create like a 'utilities' folder where we have actions like these in their own independent folder for easier readability and having a more function based repo

Reply:
Yeah, I'll do that in my code refactor story!